File: frontend/src/types/League.ts
import type { Player } from './Player';

export type League = {
  id: string;
  name: string;
  ownerId: string;
  members: number;
  position: { gameweek: number; position: number }[];
};

export type SelectedLeague = { id: string; name: string; ownerId: string; users: Player[] };

File: server/models/League.go
package models

import (
	"time"

	"gorm.io/gorm"
)

type League struct {
	// gorm.Model
	ID        string         `gorm:"primaryKey" json:"id"`
	CreatedAt time.Time      `json:"createdAt"`
	UpdatedAt time.Time      `json:"updatedAt"`
	DeletedAt gorm.DeletedAt `gorm:"index" json:"deletedAt,omitempty"`

	Name    string `json:"name"`
	OwnerID string `gorm:"not null" json:"ownerId"`
	Owner   User   `gorm:"foreignKey:OwnerID" json:"owner"`
	Users   []User `gorm:"many2many:user_languages;" json:"users"`
}
